modules/azure/monitor.go

```diff
@@ -0,0 +1,212 @@
+func DiagnosticSettingsResourceExistsE(t testing.TestingT, diagnosticSettingsResourceName string, resourceURI string, subscriptionID string) (bool, error) {
+	_, err := GetDiagnosticsSettingsResourceE(t, diagnosticSettingsResourceName, resourceURI, subscriptionID)
+	if err != nil {
+		return false, err
+	}
+	return true, nil
+}
+
+// GetDiagnosticsSettingsResource gets the diagnostics settings for a specified resource
+// This function would fail the test if there is an error.
+func GetDiagnosticsSettingsResource(t testing.TestingT, name string, resourceURI string, subscriptionID string) *insights.DiagnosticSettingsResource {
+	resource, err := GetDiagnosticsSettingsResourceE(t, name, resourceURI, subscriptionID)
+	require.NoError(t, err)
+	return resource
+}
+
+// GetDiagnosticsSettingsResourceE gets the diagnostics settings for a specified resource
+func GetDiagnosticsSettingsResourceE(t testing.TestingT, name string, resourceURI string, subscriptionID string) (*insights.DiagnosticSettingsResource, error) {
+	// Validate Azure subscription ID
+	subscriptionID, err := getTargetAzureSubscription(subscriptionID)
+	if err != nil {
+		return nil, err
+	}
+
+	client, err := GetDiagnosticsSettingsClientE(t, subscriptionID)
+
+	if err != nil {
+		return nil, err
+	}
+
+	settings, err := client.Get(context.Background(), resourceURI, name)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return &settings, nil
+}
+
+// GetDiagnosticsSettingsClientE returns a diagnostics settings client
+func GetDiagnosticsSettingsClientE(t testing.TestingT, subscriptionID string) (*insights.DiagnosticSettingsClient, error) {
+	// Validate Azure subscription ID
+	subscriptionID, err := getTargetAzureSubscription(subscriptionID)
+	if err != nil {
+		return nil, err
+	}
+
+	client := insights.NewDiagnosticSettingsClient(subscriptionID)
+	authorizer, err := NewAuthorizer()
+
+	if err != nil {
+		return nil, err
+	}
+
+	client.Authorizer = *authorizer
+	return &client, nil
+}
+
+// GetVMInsightsE get diagnostics VM onboarding status
+func GetVMInsightsE(t testing.TestingT, resourceURI string, subscriptionID string) (*insights.VMInsightsOnboardingStatus, error) {
+	client, err := GetVMInsightsClientE(t, subscriptionID)
+
+	if err != nil {
+		return nil, err
+	}
+
+	status, err := client.GetOnboardingStatus(context.Background(), resourceURI)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return &status, nil
+
+}
+
+// GetVMInsightsClientE gets a VM Insights client
+func GetVMInsightsClientE(t testing.TestingT, subscriptionID string) (*insights.VMInsightsClient, error) {
+	client := insights.NewVMInsightsClient(subscriptionID)
+
+	authorizer, err := NewAuthorizer()
+
+	if err != nil {
+		return nil, err
+	}
+
+	client.Authorizer = *authorizer
+	return &client, nil
+}
+
+// ActionGroupExists indicates whether the speficied action group exists
+func ActionGroupExists(t testing.TestingT, actionGroupName string, resGroupName string, subscriptionID string) bool {
```

Comment:
Add the error comment

Reply:
Removed the action group methods - I realized that they were redundant (there's an `actiongroup.go` file for this)